plugins/SSH.koplugin/main.lua

```diff
@@ -0,0 +1,154 @@
+                    Device.retrieveNetworkInfo and Device:retrieveNetworkInfo() or _("Could not retrieve network info.")),
+        }
+        UIManager:show(info)
+    else
+        local info = InfoMessage:new{
+                timeout = 5,
+                text = _("Error"),
+        }
+        UIManager:show(info)
+    end
+end
+
+function SSH:isRunning()
+    return util.pathExists("/tmp/dropbear.pid")
+end
+
+function SSH:stop()
+    os.execute("cat /tmp/dropbear.pid | xargs kill")
+end
+
+function SSH:show_port_dialog()
+    self.port_dialog = InputDialog:new{
+        title = _("Choose SSH port"),
+        input = self.SSH_port,
+        input_type = "number",
+        input_hint = self.SSH_port,
+        buttons = {
+            {
+                {
+                    text = _("Cancel"),
+                    callback = function()
+                        UIManager:close(self.port_dialog)
+                    end,
+                },
+                {
+                    text = _("Save"),
+                    is_enter_default = true,
+                    callback = function()
+                        local value  = tonumber(self.port_dialog:getInputText())
+                        if value then
+                            self.SSH_port = value
+                            G_reader_settings:saveSetting("SSH_port", self.SSH_port)
+                            UIManager:close(self.port_dialog)
+                        end
+                    end
+                },
+            },
+        },
+    }
+    UIManager:show(self.port_dialog)
+    self.port_dialog:onShowKeyboard()
+end
+
+function SSH:addToMainMenu(menu_items)
+    menu_items.SSH = {
+        text = _("SSH server"),
+        sub_item_table = {
+            {
+                text = _("Start SSH server"),
+                callback = function() return self:start() end,
+                enabled_func = function() return not self:isRunning() end,
+            },
+            {
+                text = _("Stop SSH server"),
+                callback = self.stop,
+                enabled_func = self.isRunning,
+            },
+            {
+                text = _("Change SSH port"),
+                enabled_func = function() return not self:isRunning() end,
+                callback = function() return self:show_port_dialog() end,
+            },
+            {
+                text = _("Login without password (DANGEROUS)"),
+                checked_func = function() return self.allow_no_password end,
+                enabled_func = function() return not self:isRunning() end,
+                callback = function() self.allow_no_password = not self.allow_no_password end,
+            },
+            {
+                text = _("SSH public key"),
+                callback = function()
+                    local info = InfoMessage:new{
+                        timeout = 5,
+                        text = _("Put your public SSH keys in settings/SSH/authorized_keys"),
```

Comment:
Should we give the full path?

Reply:
I just found lfs.currentdir().
Yes that's better with the full path.